File: server/src/main/java/com/example/server/model/DiaryEvent.java
package com.example.server.model;

import jakarta.persistence.*;
import lombok.Data;

@Entity
@Data
@Table(name = "diary_events")
public class DiaryEvent {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private Long userId;
    private Long partnerId;

    private String title;

    @Column(columnDefinition = "TEXT")
    private String description;

    private String imageSmallUrl;
    private String imageLargeUrl;

    private Long date; // epoch millis

    @Enumerated(EnumType.STRING)
    private EventType type;
}

File: client/app/src/main/java/com/example/us/ui/screen/diary/DiaryEvent.kt
package com.example.us.ui.screen.diary

import kotlinx.serialization.Serializable

@Serializable
data class DiaryEvent(
    val id: Int,
    val title: String,
    val description: String,
    val date: Long,
    val imageSmallUrl: String?,
    val imageLargeUrl: String?,
    val type: EventType
)

enum class EventType {
    HAPPY, ROMANTIC, SAD, IMPORTANT, FUNNY, SPICY_18
}
